### ISSUE TITLE
Revise application startup in server.ts?

### ISSUE THREAD

**Issue by lawlsausage** [stvcott8511/monitor#21]
In reference to #20, should we revise the application startup in server.ts so it is structured to bring configurations into memory before starting the server? At the moment, it structurally looks a little confusing with the imports and order of invoking the .env config. Maybe we can revise this to have a main.ts that starts the application which brings in pre-configurations before starting the server. The other option is to instead maybe implement the config as a dependency injection through perhaps a simple IOC container.

**Comment by stvcott8511** [comment]
I agree that change is needed. However, to keep master in a working state I made the change. I will create and issue to clean up settings.

**Comment by lawlsausage** [comment]
Actually, there's also the following from the dotenv project:

![image](https://user-images.githubusercontent.com/6690828/76154164-c0477400-60a5-11ea-82fa-5ac669fc2a11.png)

https://github.com/motdotla/dotenv

We can probably just get rid of the line entirely and add the CLI arguments to the package.json scripts.

**Comment by stvcott8511** [comment]
@lawlsausage The Preload will not work with the Webpack unless I'm missing something with the Webpack process that will preserve some of the node_modules.

**Comment by lawlsausage** [comment]
That makes sense since webpack will transpile for the environment being built. I found the following possibilities:

- https://stackoverflow.com/questions/46224986/how-to-pass-env-file-variables-to-webpack-config
- https://www.npmjs.com/package/dotenv-webpack (probably should use the --save-dev/-D argument instead)

I don't really have a preference to either... I think the first would benefit from requiring less external dependencies versus the second option.

**Comment by lawlsausage** [comment]
Just interesting, but create-react-app has a pretty good way of how they handle dotenv files in my opinion. We don't need an immediately robust solution, but this might be some good inspiration.

https://github.com/facebook/create-react-app/search?q=dotenv&unscoped_q=dotenv

**Comment by lawlsausage** [comment]
@stvcott8511 

I think in regards to docker and the direction you're looking for, it almost looks like we don't necessarily want env variable values read and transpiled into the build. The env files will need to be read in at application start. That means that create-react-app's approach is not quite what we need. Maybe your initial approach was actually going in the right direction, but I think that we might need to delay the initialization of many of the modules until configuration is read in.

- Limit invocation of module functions in the global scope.
- Single entry point from the global scope to start the rest of the application (like a main function).
- Consider DI and maybe a light IOC.

**Comment by lawlsausage** [comment]
@stvcott8511 

I think it's best if the rest of the modules are migrated to TypeScript before taking this on. Refactoring to account for the above points can happen during the migration. Each change can then be compartmentalized.
